v4/newrelic/segments.go

```diff
@@ -170,7 +170,15 @@ func (s *span) isEnded() bool {
 //
 // The key must contain fewer than than 255 bytes.  The value must be a
 // number, string, or boolean.
-func (s *Segment) AddAttribute(key string, val interface{}) {}
+func (s *Segment) AddAttribute(key string, val interface{}) {
+	if s == nil {
+		return
+	}
+	if s.StartTime.span == nil {
+		return
+	}
+	s.StartTime.Span.SetAttribute(key, val)
```

Comment:
This doesn't actually have to do with this PR, it's just something that's been floating around in my head: I don't like the name `StartTime`.  It seems counter intuitive to what it actually represents. I'm guessing we need to keep it that way because it is public and we don't want to change the API, but if there is any way as we work on this shim work that we can find to change it to something a little more intuitive, I would like that.

Reply:
Agreed, I've been feeling the same thing. But yeah, the problem is that it's public so we can't change the name.